x-pack/plugins/security_solution/server/lib/detection_engine/notifications/schedule_notification_response_actions.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { map, uniq } from 'lodash';
+import type { SetupPlugins } from '../../../plugin_contract';
+
+interface OsqueryQuery {
+  id: string;
+  query: string;
+  ecs_mapping: Record<string, Record<'field', string>>;
+  version: string;
+  platform: string;
+}
+
+interface OsqueryResponseAction {
+  actionTypeId: string;
+  params: {
+    id: string;
+    queries: OsqueryQuery[];
+    savedQueryId: string;
+    query: string;
+    packId: string;
+    ecs_mapping?: Record<string, { field?: string; value?: string }>;
+  };
+}
+
+export type ResponseAction = OsqueryResponseAction;
+
+interface ScheduleNotificationActions {
+  signals: unknown[];
+  responseActions: ResponseAction[];
+}
+
+interface IAlert {
+  agent: {
+    id: string;
+  };
+}
+
+export const scheduleNotificationResponseActions = (
+  { signals, responseActions }: ScheduleNotificationActions,
+  osqueryCreateAction?: SetupPlugins['osquery']['osqueryCreateAction']
+) => {
+  const filteredAlerts = (signals as IAlert[]).filter((alert) => alert.agent?.id);
+  const agentIds = uniq(filteredAlerts.map((alert: IAlert) => alert.agent?.id));
+  const alertIds = map(filteredAlerts, '_id');
+
+  responseActions.forEach((responseActionParam) => {
+    if (responseActionParam.actionTypeId === '.osquery' && osqueryCreateAction) {
+      const {
+        savedQueryId,
+        packId,
+        queries,
+        ecs_mapping: ecsMapping,
+        ...rest
+      } = responseActionParam.params;
+      return osqueryCreateAction({
+        pack_id: packId,
+        queries: map(queries, (query, queryId: string) => {
+          return {
+            ...query,
+            id: queryId,
+            query: query.query,
+            ecs_mapping: query.ecs_mapping,
+          };
+        }) as unknown as OsqueryQuery[],
```

Comment:
This cast looks strange to me, how do we know that the type of this `map` will be `OsqueryQuery[]` even though TS says the return type is `boolean[]`? I couldn't find a reference to how the second argument `queryId` in the map callback is supposed to work, and removing that argument fixes the return type - seems like there might be a bug here.

Reply:
Fixed, thank you, the ID was not expected.